superpeer/handler/MenuHandler.py

```diff
@@ -11,21 +37,270 @@ def serve(self, choice: str) -> None:
 		"""
 
 		if choice == "SUPE":
-			pass
+			# SUPE[4B].Packet_Id[16B].IP_Peer[55B].Port_Peer[5B].TTL[2B]
+			pktid = str(uuid.uuid4().hex[:16].upper())
+			ip = net_utils.get_local_ip_for_response()
+			port = net_utils.get_asup_port()
+			ttl = '04'
+
+			packet = choice + pktid + ip + str(port).zfill(5) + ttl
+			LocalData.set_sent_packet(pktid)
+
+			old_superfriends_len = len(LocalData.get_super_friends())
+
+			server = ServerThread(port, TimedResponseHandler())
+			server.daemon = True
+			server.start()
+
+			spinner = SpinnerThread('Searching peers (ENTER to continue)', 'Research done!')
+			spinner.start()
+
+			timer = Timer(20, lambda: (spinner.stop(), server.stop()))
+			timer.start()
+
+			self.__broadcast(packet)
+			input()
+			print('\033[1A', end='\r')
+
+			if timer.is_alive():
+				spinner.stop()
+				spinner.join()
+				timer.cancel()
+				timer.join()
+				server.stop()
+				server.join()
+			else:
+				spinner.join()
+				timer.join()
+				server.join()
+
+			if len(LocalData.get_super_friends()) == old_superfriends_len:
+				shell.print_red('\nNo new superpeer found.\n')
+			else:
+				for friend in LocalData.get_super_friends():
+					superfriend_ip4 = LocalData.get_super_friend_ip4(friend)
+					superfriend_ip6 = LocalData.get_super_friend_ip6(friend)
+					superfriend_port = LocalData.get_super_friend_port(friend)
+					shell.print_green(
+						f'\nNew superpeer found: {LocalData.super_friend_index(superfriend_ip4,superfriend_ip6,superfriend_port)})' 
+						f' {superfriend_ip4}|{superfriend_ip6} on port {superfriend_port}')
```

Comment:
Per me sarebbe più bello se la stampa di ```New superpeer found...``` la facesse il TimedResponseHandler ad ogni ASUP ricevuta come in R00tella..che dite? @GiovanniFiorini @jhonrambo93 @xBlue0